c/zos.c

```diff
@@ -108,6 +108,7 @@ int supervisorMode(int enable){
 }
 
 int setKey(int key){
+#ifndef __ZOWE_COMP_XLCLANG  /* temp hack until Joe figures out what "&r" means */
```

Comment:
> /* temp hack until Joe figures out what "&r" means */

From GCC:
> Means (in a particular alternative) that this operand is an earlyclobber operand, which is modified before the instruction is finished using the input operands. Therefore, this operand may not lie in a register that is used as an input operand or as part of any memory address. 

I think by default the compiler assumes that the input register aren't modified before the corresponding instruction have read the input values, which is not true in case of `SLL` which changes `%1` followed by `MODESET` reading `%1`. Because of that assumption the compiler may make some optimizations which will break the code.

We should probably change `return 8;` to actually set the key, or maybe abort, so it works in a predictable way.

Reply:
Is the use of &r really needed for the non-xlclang version.  I wrote this originally and didn't use "&r" and it worked.

Reply:
It's hard to tell, I can ask the author of `&` in `&r`.

Reply:
Can we just abort until this gets implemented then?

Reply:
I will test a fix that gets rid of the &.    I don't know who made this change but will find out.

Reply:
Joe fixed this by moving the left shift to the C code and removing `&`.